### ISSUE TITLE
mat-menu focuses the second item on Arrow Down when opened with the mouse

### ISSUE THREAD

**Issue by JoshuaEN** [angular/components#16184]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
When opening a menu with the mouse, no items are focused. Pressing arrow down (on the keyboard) in this state (where no item is focused) would focus the first item; pressing arrow up would focus the last item (via wrapping around).

#### What is the current behavior?
Opening a menu with the mouse and then pressing arrow down focuses the second menu item.

#### What are the steps to reproduce?
1. Go to
    * https://material.angular.io/components/menu/overview
    * or https://v7.material.angular.io/components/menu/overview
    * or https://v6.material.angular.io/components/menu/overview
2. Click on the "Menu" button in the basic menu example.
3. Press Down Arrow on the keyboard

Recording: [2019-06-02_19-34-41](https://user-images.githubusercontent.com/5409190/58768724-7d33ff80-856d-11e9-9496-452b95939d56.gif)

#### What is the use-case or motivation for changing an existing behavior?
The current behavior is non-intuitive: Nothing is visibility focused, but the menu behaves as if the first item is focused.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular Material 6.4.7, 7.3.7, and 8.0.0
Tested in Firefox and Chrome on Windows 10

#### Is there anything else we should know?
Root cause appears to be that there is always a menu item focused; but, the focused menu item when the menu is opened by the mouse is styled like a normal menu item, giving the impression there is no item focused by default.

This can be observed (at least for 8.0.0) by adding a style for `.cdk-focused.cdk-mouse-focused`, and observing that the style is applied to the first menu item when opening the menu with the mouse.

---

Potentially related issue: https://github.com/angular/components/issues/2657

That issue describes the behavior I see in 5.2.5 (here: https://v5.material.angular.io/components/menu/overview ), where opening the menu with the mouse disables focus styling (making keyboard navigation hard to use), but opening the menu with the keyboard allows for keyboard navigation (whereas now, the focus state is hidden until the user interacts with the keyboard). At the time, the issue was closed saying the behavior was intended.

**Comment by crisbeto** [comment]
@jelbourn currently this is intentional. Focus is actually on the first item, but we don't show the focus indication unless the menu was opened using the keyboard.

**Comment by jelbourn** [comment]
Right, I remember now. I'd be curious to see what other menu implementations do

**Comment by JoshuaEN** [comment]
> I'd be curious to see what other menu implementations do

I went through a few websites/applications to see:

Behavior as described in "Expected behavior":
* Gmail [[Email context menu](https://user-images.githubusercontent.com/5409190/59151815-9aeae400-8a07-11e9-825d-183fb63c2b85.gif)], [[Tool bar ... menu](https://user-images.githubusercontent.com/5409190/59151819-a76f3c80-8a07-11e9-9a7f-7675a0c55621.gif)]
* Chrome [[Context menu](https://user-images.githubusercontent.com/5409190/59151901-cae6b700-8a08-11e9-8808-74d915773355.gif)], [[Browser ... menu](https://user-images.githubusercontent.com/5409190/59151915-ec47a300-8a08-11e9-862b-a3fb5ad984b0.gif)], [[Bookmark ... menu](https://user-images.githubusercontent.com/5409190/59151931-33359880-8a09-11e9-91c5-f331c4e5a120.gif)]
* GitHub [[Post ... menu](https://user-images.githubusercontent.com/5409190/59151977-b48d2b00-8a09-11e9-8756-3e7a8b4d057f.gif)], [[Header new menu](https://user-images.githubusercontent.com/5409190/59151986-e69e8d00-8a09-11e9-9edb-0efe256e5d73.gif)], 
* Windows Task Manager [[Services tab](https://user-images.githubusercontent.com/5409190/59152013-89570b80-8a0a-11e9-9ee7-3510393b661f.gif)]
* Windows File Explorer [[File & Folder context menu](https://user-images.githubusercontent.com/5409190/59152028-c3c0a880-8a0a-11e9-85fb-d535b067e77b.gif)]
* Windows [[Program menu](https://user-images.githubusercontent.com/5409190/59152061-17cb8d00-8a0b-11e9-961d-5e5d3bcc22f9.gif)]
* FireFox [[Context menu](https://user-images.githubusercontent.com/5409190/59152109-b3f59400-8a0b-11e9-8d40-4ba3a5f1b1d4.gif)], [[Browser ... menu](https://user-images.githubusercontent.com/5409190/59152123-e2736f00-8a0b-11e9-910c-3db4f2ac6f43.gif)]
* Microsoft Word [[Editor context menu](https://user-images.githubusercontent.com/5409190/59152141-2c5c5500-8a0c-11e9-950e-29f553b518d9.gif)], [[Find more options menu button](https://user-images.githubusercontent.com/5409190/59152145-4dbd4100-8a0c-11e9-9691-b963912d9474.gif)]
* Microsoft Excel [[Cell context menu](https://user-images.githubusercontent.com/5409190/59152155-89f0a180-8a0c-11e9-8df6-16c5dc78a7be.gif)], [[AutoSum more options menu button](https://user-images.githubusercontent.com/5409190/59152158-9d9c0800-8a0c-11e9-93ce-77f4afb4485f.gif)]
* ShareX [[Capture menu](https://user-images.githubusercontent.com/5409190/59152165-d3d98780-8a0c-11e9-9334-eadc088f06a1.gif)]
* Google Docs [[File ... menu](https://user-images.githubusercontent.com/5409190/59152172-1dc26d80-8a0d-11e9-99e0-dc57f5875118.gif)], [[File context menu](https://user-images.githubusercontent.com/5409190/59152182-4480a400-8a0d-11e9-93b4-b161db0aea96.gif)], [[New menu](https://user-images.githubusercontent.com/5409190/59152200-6ed26180-8a0d-11e9-98f0-c1d45faeea62.gif)]
* Slack [[Channel toolbar menus](https://user-images.githubusercontent.com/5409190/59152251-6890b500-8a0e-11e9-84e8-0a8b99cdb2af.gif)], [[Message ... menu](https://user-images.githubusercontent.com/5409190/59152262-89590a80-8a0e-11e9-96ec-b1695683c743.gif)]
* VSCode [[Menu bar menus](https://user-images.githubusercontent.com/5409190/59152286-dfc64900-8a0e-11e9-9eda-564ba1bd8986.gif)], [[Context menu](https://user-images.githubusercontent.com/5409190/59152297-02f0f880-8a0f-11e9-96c1-471c6f018616.gif)], [[Tab list ... menu](https://user-images.githubusercontent.com/5409190/59152302-1bf9a980-8a0f-11e9-9c07-ca88998d18b1.gif)]

Behavior same as current behavior in Angular:
* Edge [[Context menu](https://user-images.githubusercontent.com/5409190/59152077-53feed80-8a0b-11e9-9d08-3c9c3989404d.gif)], [[Browser ... menu](https://user-images.githubusercontent.com/5409190/59152088-7e50ab00-8a0b-11e9-9817-c05f896d89b1.gif)]
* Microsoft Teams [[User menu](https://user-images.githubusercontent.com/5409190/59152218-d4bee900-8a0d-11e9-9830-75a9efc835bd.gif)], [[Message ... menu](https://user-images.githubusercontent.com/5409190/59152236-1485d080-8a0e-11e9-8528-d88245720f66.gif)]

---

Other behaviors:

Behavior of always showing a selected item:
* Gmail [[Help menu](https://user-images.githubusercontent.com/5409190/59151842-06cd4c80-8a08-11e9-9749-22bc3ce2e5f3.gif)]

Behavior of always selecting the first item upon keyboard input (1st arrow up press also selects the first item, rather than the last item):
* Visual Studio 2019 [[Menu bar menus](https://user-images.githubusercontent.com/5409190/59152322-86aae500-8a0f-11e9-8bee-06779ef4e4d0.gif)]

---

Notes about methodology:

* I noted the behavior of all applications/websites I tested, except if it did not support menu keyboard interaction.
* I did not provide recording demos of menus within an application/website I tried, only a few samples and any cases that diverged (e.g. Gmail's help menu).
* I tried to always include the on-screen keyboard, but some applications did not send on-screen keyboard events to the open context menu.

**Comment by paulirwin** [comment]
> > I'd be curious to see what other menu implementations do
> 
> I went through a few websites/applications to see:
> 
> Behavior as described in "Expected behavior":

To add some more to the list: macOS and iPadOS (with external keyboard) native menus do this expected behavior. Opening a menu from the top bar via mouse (or tap), or from a right click (or long press) menu focuses the first or last item based on up or down press. 

(iPadOS has some strange behavior where no matter whether you press up or down keys from a long-press menu like a dock icon, the top item is always selected. This differs from macOS, where it will select either the first or last item based on which key was used.)

Gnome on Linux visibly pre-selects the first menu item when opened.

I am fine with either of these behaviors, but the current Angular experience is very strange and uncommon. To select the first item via keyboard after opening it with a mouse, you have to press down and then up. At the very least, I would like an option to be able to change this behavior to what is described in this issue.
